### PR TITLE
Don't update nodes list when stepping down as master

### PR DESCRIPTION
**Pull request by ywelsch**
This commit simplifies the node update logic so that nodes are never removed from the cluster state when the cluster state is not published.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java [hunk 1]
```diff
@@ -505,12 +506,27 @@ private boolean joinElectedMaster(DiscoveryNode masterNode) {
         }
     }
 
+    private void submitRejoin(String source) {
```
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java]: shall we use this in org.elasticsearch.discovery.zen.ZenDiscovery#publish as well?

**Review by bleskes — APPROVED** [review]
LGTM

File: core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java (hunk 1 above)
Comment (ywelsch) [inline core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java]: good idea

**Comment by ywelsch** [comment]
Thanks @bleskes

**Comment by ywelsch** [comment]
backported to 5.2.0

**Pull request merged**